Client/Frontend/Menu/AppMenuConfiguration.swift

```diff
@@ -112,7 +116,11 @@ struct AppMenuConfiguration: MenuConfiguration {
             if #available(iOS 9, *) {
                 menuItems.append(AppMenuConfiguration.NewPrivateTabMenuItem)
             }
-            menuItems.append(tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem)
+            var menuItem = tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem
+            if let errorPageUrl = tabState.url where ErrorPageHelper.isErrorPageURL(errorPageUrl) {
```

Comment:
Sorry, missed your question!

It looks like this function does still take a parameter; see [here](https://github.com/mozilla/firefox-ios/blob/aae3fdc32c7a11bd5fd7aa32ba457651e0a278e9/Utils/Extensions/NSURLExtensions.swift#L246). Perhaps your branch is outdated?

> I assume includeDataURIs: false makes us exclude reader mode pages

Reader mode pages are actually pages on the local web server that we run alongside the browser. We don't use data URIs for anything on Firefox for iOS, but the user may navigate to a data URI manually. Thinking about it more, we may want to pass `true` here instead of false; I can't think of a reason why we shouldn't be able to bookmark a data URI (and desktop allows it).

> How come we still display the actual http url in the address bar though?
A reader mode URL is something like this: `http://localhost:1234/reader-mode/reader.html?url=http://www.example.com`. That's not a very attractive URL, and there's no reason to expose the `localhost` URL to the user, so we simply show the URL of the original page (even though that's not the actual URL of the content).